src/sync/mpsc/mod.rs

```diff
@@ -459,6 +459,23 @@ impl<T> Sender<T> {
         self.maybe_parked = state.is_open;
     }
 
+    /// Polls the channel to determine if there is guaranteed to be capacity to send at least one
+    /// item without waiting.
+    ///
+    /// Returns `Ok(Async::Ready(_))` if there is sufficient capacity, or returns
+    /// `Ok(Async::NotReady)` if the channel is not guaranteed to have capacity. Returns `Err(_)`
+    /// if the receiver has been dropped.
+    ///
+    /// This method may only be called from inside the context of a task or future.
+    pub fn poll_ready(&mut self) -> Poll<(), ()> {
```

Comment:
Perhaps the error here could be some form of `SendError` to have roughly the same error type as the signatures below?

Reply:
Yes, the signature could change to be `pub fn poll_ready(&mut self) -> Poll<(), SendError<()>>;`, I went with plain `()` to keep it simple, since `SendError<()>` is pretty much equivalent, but if you prefer `SendError<()>` I can switch it over.